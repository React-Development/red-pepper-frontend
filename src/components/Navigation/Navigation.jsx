import React, { useState } from "react";
import AppBar from "@material-ui/core/AppBar";
import Divider from "@material-ui/core/Divider";
import Drawer from "@material-ui/core/Drawer";
import Hidden from "@material-ui/core/Hidden";
import IconButton from "@material-ui/core/IconButton";
import List from "@material-ui/core/List";

import MenuIcon from "@material-ui/icons/Menu";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { makeStyles, useTheme } from "@material-ui/core/styles";

import { grey, red } from "@material-ui/core/colors";
import DrawerOptions from "./utils/DrawerOptions";
import { LogOutButton } from "../UI/Buttons/Buttons";

// Styles
const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
  },
  drawer: {
    [theme.breakpoints.up("sm")]: {
      width: drawerWidth,
      flexShrink: 0,
    },
  },
  appBar: {
    backgroundColor: red[500],
    zIndex: theme.zIndex.drawer + 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up("sm")]: {
      display: "none",
    },
  },
  // necessary for content to be below app bar
  toolbar: theme.mixins.toolbar,
  drawerContainer: {
    overflow: "auto",
  },
  drawerPaper: {
    width: drawerWidth,
    backgroundColor: grey[50],
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
  title: {
    flexGrow: 1,
  },
}));

// Main Component
const Navigation = (props) => {
  const { window } = props;
  const classes = useStyles();
  const theme = useTheme();
  const [mobileOpen, setMobileOpen] = useState(false);
  const { isAuthenticated, location } = props;

  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };

  const drawer = (
    <div>
      <div className={classes.toolbar} />
      <Divider />
      <List>
        <DrawerOptions location={location} />
      </List>
    </div>
  );

  const container =
    window !== undefined ? () => window().document.body : undefined;

  return (
    <div className={classes.root}>
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          {isAuthenticated ? (
            <IconButton
              color="inherit"
              aria-label="open drawer"
              edge="start"
              onClick={handleDrawerToggle}
              className={classes.menuButton}
            >
              <MenuIcon />
            </IconButton>
          ) : null}

          <Typography variant="h6" noWrap className={classes.title}>
            Red Pepper App
          </Typography>
          {isAuthenticated ? (
            <LogOutButton onClick={props.onLogout}>Cerrar Sesi√≥n</LogOutButton>
          ) : null}
        </Toolbar>
      </AppBar>

      {isAuthenticated ? (
        <nav className={classes.drawer} aria-label="menu options">
          {/* Mobile Version */}
          <Hidden smUp implementation="css">
            <Drawer
              container={container}
              variant="temporary"
              anchor={theme.direction === "rtl" ? "right" : "left"}
              open={mobileOpen}
              onClose={handleDrawerToggle}
              classes={{
                paper: classes.drawerPaper,
              }}
              ModalProps={{
                keepMounted: true, // Better open performance on mobile.
              }}
            >
              {drawer}
            </Drawer>
          </Hidden>

          {/* Desktop Version */}
          <Hidden xsDown implementation="css">
            <Drawer
              classes={{
                paper: classes.drawerPaper,
              }}
              variant="permanent"
              open
            >
              {drawer}
            </Drawer>
          </Hidden>
        </nav>
      ) : null}
    </div>
  );
};

export default Navigation;
