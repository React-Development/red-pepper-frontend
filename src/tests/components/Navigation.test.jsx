import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Navigation from "components/Navigation/Navigation";

test("should render header correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Navigation />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
