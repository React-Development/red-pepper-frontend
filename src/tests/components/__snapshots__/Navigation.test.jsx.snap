// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <ForwardRef(WithStyles)
    className="makeStyles-appBar-3"
    position="fixed"
  >
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-title-9"
        noWrap={true}
        variant="h6"
      >
        Red Pepper App
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</div>
`;
